### PR TITLE
[6CORE-026] create REST Specification for Warehouse Service: Product Exploration

### PR DESCRIPTION
**Pull request by kateryna-mykh**

### CODE REVIEW

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductCartResponse.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com._6core.platform.warehousespec.rest.v1.dto.product;
+
+import java.math.BigInteger;
+
+public record ProductCartResponse(
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductCartResponse.java]: Cart? There should no cart-related info be in this service.
```suggestion
public record ProductCartResponse(
```

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/ProductGetApi.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    return Flux.just(ResponseEntity.ok().body(mockCartResponse));
+  }
+
+  /**
+   * Exploring exact product with all details GET /products/{productId}/details
+   *
+   * @param productId (required)
+   * @return OK (status code 200)
+   */
+  @GetMapping(value = "/products/{productId}/details", produces = "application/json")
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/ProductGetApi.java]:
```suggestion
  @GetMapping(value = "/products/{productId}", produces = "application/json")
```

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductCartResponse.java (hunk 1 above)
Comment (kateryna-mykh) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductCartResponse.java]: @sokimaaa, Is ProuductCardResonse better?

Comment (sokimaaa) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductCartResponse.java]: Card is a flat stiff usually small and rectangular piece of material (such as paper, cardboard, or plastic) usually bearing information. e.g. Credit Card.

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductSearchResponseCard.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com._6core.platform.warehousespec.rest.v1.dto.product;
+
+import java.math.BigInteger;
+
+public record ProductSearchResponseCard(
```
Comment (sokimaaa) [inline 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductSearchResponseCard.java]: 1. `Response` should be in the end according to our naming style. 
2. Card is about a flat stiff usually small and rectangular piece of material (such as paper, cardboard, or plastic) usually bearing information, e.g. Credit Card.

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductSearchResponseCard.java (hunk 3 above)
Comment (kateryna-mykh) [reply 3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/dto/product/ProductSearchResponseCard.java]: @sokimaaa, I think about this response object as 'product card' - UI element that shows short info about a product.
image

**Pull request merged**